source/aws_s3/s3.go

```diff
@@ -21,37 +21,72 @@ func init() {
 
 type s3Driver struct {
 	s3client   s3iface.S3API
-	bucket     string
-	prefix     string
+	config     *Config
 	migrations *source.Migrations
 }
 
+type Config struct {
+	Bucket string
+	Prefix string
+}
+
 func (s *s3Driver) Open(folder string) (source.Driver, error) {
```

Comment:
Why not call `WithInstance()` in `Open()`?

Reply:
Good idea

Reply:
Done